drake/core/Core.h

```diff
@@ -100,18 +109,19 @@ namespace Drake {
   template <typename ScalarType, template <typename> class Vector1, template <typename> class Vector2>
   class CombinedVector {
   public:
-    const static size_t vec1_rows = Vector1<ScalarType>::RowsAtCompileTime,
-                        vec2_rows = Vector2<ScalarType>::RowsAtCompileTime;
     CombinedVector() {};  // allow use of default constructors for vec1 and vec2, also
     CombinedVector(const Vector1<ScalarType>& first, const Vector2<ScalarType>& second) : vec1(first), vec2(second) {};
 
-    template <typename Derived> CombinedVector(const Eigen::MatrixBase<Derived>& x) : vec1(x.topRows(vec1_rows)), vec2(x.bottomRows(vec2_rows)) {};
+    template <typename Derived> CombinedVector(const Eigen::MatrixBase<Derived>& x) : vec1(x.topRows(Vector1<ScalarType>::RowsAtCompileTime)), vec2(x.bottomRows(Vector2<ScalarType>::RowsAtCompileTime)) {
+      static_assert(RowsAtCompileTime != Eigen::Dynamic, "Cannot determine sizes of subvectors because sizes are not known at compile time."); // TODO: could handle cases where only one of the subvectors has dynamic size
+    };
```

Comment:
I don't really like that because it means that code that calls this is by design not going to work for certain state vectors that do satisfy our vector concept.